### PR TITLE
CI workflow on Windows, macOS, and Ubuntu

### PR DESCRIPTION
**Pull request by eggplants**
To check compatibility of major environments, I modified CI and test files so that they could be run on mac and windows.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -4,31 +4,49 @@ on: [push, pull_request]
...
         python -m pip install wheel
         python setup.py bdist_wheel
-        cd dist
-        python -m pip install flake8_black-*.whl
-        cd ..
-        # Wheel should now be installed
-        flake8 --version
-    - name: Run tests
+        ls dist
+    - name: Check if Manifest was complete
```
Comment (peterjc) [inline .github/workflows/test.yml:44]: The way we check the manifest was complete enough to run the tests is to unpack it and run the tests on the unpacked files. That functionality has been lost here (perhaps it was not explicit enough in the comments?)

File: tests/run_tests.sh [hunk 2]
```diff
@@ -36,17 +36,21 @@ flake8 --select BLK conflicting_configurations/*.py
 flake8 --select BLK with_bad_toml/hello_world.py --black-config ''
 
 echo "Checking we report expected black changes"
-diff test_changes/hello_world.txt <(flake8 --select BLK test_changes/hello_world.py)
-diff test_changes/hello_world_EOF.txt <(flake8 --select BLK test_changes/hello_world_EOF.py)
-diff test_changes/hello_world_EOF.txt <(flake8 --select BLK test_changes/hello_world_EOF.py --black-config '')
-diff with_bad_toml/hello_world.txt <(flake8 --select BLK with_bad_toml/hello_world.py)
-diff with_pyproject_toml/ignoring_toml.txt <(flake8 with_pyproject_toml/ --select BLK --black-config '')
+diff --strip-trailing-cr test_changes/hello_world.txt <(flake8 --select BLK test_changes/hello_world.py)
```
Comment (peterjc) [inline tests/run_tests.sh:39]: Can we not tweak the Windows git ``core.autocrlf`` newline settings instead?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -4,31 +4,49 @@ on: [push, pull_request]
...
     - uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Build packages, and install the wheel
+    - name: Install packages
+      run: |
+        pip install -U pip
+        pip install .
```
Comment (peterjc) [inline .github/workflows/test.yml:22]: This just tests installing from source works - I wanted to explicitly test the wheel would work.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -39,5 +43,10 @@ jobs:
         ls dist
     - name: Check if Manifest was complete
       run: |
-        tar -zxvf dist/flake8-black-*.tar.gz
+        if [ "$RUNNER_OS" == "Windows" ]; then
+          tar zxvf dist/flake8-black-*.tar.gz
+        else
+          tar -zxvf dist/flake8-black-*.tar.gz
+        fi
```
Comment (peterjc) [inline .github/workflows/test.yml:50]: No need for the if statement? Linux and macOS should accept either ``tar zxvf ...`` or ``tar -zxvf ...``

File: .github/workflows/test.yml (hunk 3 above)
Comment (peterjc) [reply .github/workflows/test.yml:22]: Which you do again later (seems redundant?)

File: tests/run_tests.sh (hunk 2 above)
Comment (peterjc) [reply tests/run_tests.sh]: I was thinking if git is told to do the line ending conversion automatically, then plain ``diff`` should just work even on Windows?

**Comment by peterjc** [comment]
That's looking much cleaner now :)

**Comment by peterjc** [comment]
Got the plain diff to work - see https://github.com/peterjc/flake8-black/commits/win_mac_ci

It seems ``git config core.autocrlf false`` was needed but I don't understand why. Perhaps it has to do with running bash on Windows?

**Comment by eggplants** [comment]
>Got the plain diff to work - see https://github.com/peterjc/flake8-black/commits/win_mac_ci
>
>It seems git config core.autocrlf false was needed but I don't understand why. Perhaps it has to do with running bash on Windows?

!!!
I haven't known this behavior! I cannot explain why too.
Could you add your commits to here?

**Comment by peterjc** [comment]
I was expecting ``git config core.autocrlf auto`` or ``git config core.autocrlf true`` to be appropriate, but that didn't work. I didn't expect ``git config core.autocrlf false`` to work but tried it anyway.

I can try updating this branch...

**Comment by peterjc** [comment]
Rebased, and trying without the ``git reset --hard`` (might need that too)

**Comment by peterjc** [comment]
Are you OK with the modified branch being merged?

(You can force push your original if you'd rather we keep more of the history)

**Comment by eggplants** [comment]
> Are you OK with the modified branch being merged?
> 
> (You can force push your original if you'd rather we keep more of the history)

Ready for merge.

**Comment by peterjc** [comment]
Merged, thank you!

**Pull request merged**